src/rustdoc-json-types/lib.rs

```diff
@@ -562,6 +562,12 @@ pub enum Type {
         type_: Box<Type>,
         len: String,
     },
+    /// `u32 is 1..`
+    Pat {
+        #[serde(rename = "type")]
+        type_: Box<Type>,
+        pat: String,
```

Comment:
Would it be possible to document what `pat` here holds?

Is it `"1.."`? Is it `"u32 is 1.."`?

Right now it seems like downstream users (like `cargo-semver-checks`) would be expected to parse this pattern before making use of it. It'd be great to specify its format to avoid needing a trial-and-error approach downstream.

Reply:
I renamed the field to `__pat_unstable_do_not_use` 😆

Reply:
Consider `#[doc(hidden)]` in that case? I think that might better communicate the "this is not public API yet, breakage will happen often" intent.

For example, right now `cargo-semver-checks` will consider the field public API. But if you mark it `#[doc(hidden)]`, it'll be considered non-public API even though it's a public field of a public variant.

Reply:
I thought you just processed the json. Are you using the rustdoc sources directly?

Reply:
Sorry, I think we might not be talking about the same thing.

`cargo-semver-checks` uses the `rustdoc-types` crate, which just is the crates.io version of this code. So it parses the JSON with essentially using this same code. But it won't _rely on_ this field, because I haven't written any code that uses it and I probably won't anytime soon given the unstable nature of it.

I meant "will consider it public API" in the sense of "if one scans this code with `cargo-semver-checks`" though. To me, `#[doc(hidden)]` makes sense in the "don't use this, it's unstable" sense aimed at humans, and would also give the correct impression to tools like rustdoc and `cargo-semver-checks`.

(Side note in case it's relevant: the "is public API" algorithm in `cargo-semver-checks` is basically "does reaching the item _require_ passing through `#[doc(hidden)]` anywhere" because that accounts for cases like:
```rust
#[doc(hidden)]
pub mod inner {
    pub fn item() {}
}

pub use inner::item;
```
where `item` is public API since its use as `this_crate::item` is not `#[doc(hidden)]`.)